subjects/mobile-dev/secure-messenger/README.md

```diff
@@ -1,36 +1,30 @@
-
-To enable user authentication implement Firebase Authentication and secure sign-in. Users should be able to sign up, log in, and log out of the app using email and password authentication. Additionally, implement biometric authentication using the local_auth package to allow users to log in using their fingerprint or face ID.
-
-Implement end-to-end encryption for secret chats using the encrypt or webcrypto packages. Use a symmetric encryption algorithm like RSA, and generate a new public and private key pair for each pair of users for each secret chat. The public keys should be securely shared between the sender and receiver.
-
-### Implement the following features:
-
-- User profiles: Users should be able to create a profile with a profile picture, username, and other information.
-- Search users: Users should be able to search for other users by username or other information.
-- Add contacts: Users should be able to add contacts by searching for their username or scanning a QR code.
-- Messaging: Users should be able to send text messages, images, and videos, with features such as read receipts, typing indicators, and the ability to edit or delete messages.
-- Secret chats: Users should be able to start encrypted one-on-one chats with the same functionality as regular chats with one exception: all messages have to be encrypted using an end-to-end algorithm.
+- **Login/Signup**: Implement a login/signup page and the necessary functionality for user authentication and account creation.
+- **User profile**: Users must be able to create a profile with a profile picture, username, and other information.
+- **Search users**: Users must be able to search for other users by username or other information.
+- **Add contacts**: Users must be able to add contacts by searching for their username or scanning a QR code.
+- **Messaging**: Users must be able to send text messages, images, and videos. Apart from that you must implement these features:
+  - **read receipts**
+  - **typing indicators**
+  - **edit** or **delete** messages.
+- **Secret chats**: Users must be able to start encrypted one-on-one chats with the same functionality as regular chats with one exception: all messages have to be encrypted using an end-to-end algorithm.
 
 ### Bonus:
 
-Group messaging: Users should be able to create or join group chats with multiple users having at least the same functionality as regular chats.
+- **Group messaging**, where users are able to create or join group chats with multiple users having at least the same functionality as regular chats.
 
-Push notifications: Users should receive notifications of new messages or friend requests using the firebase_messaging package.
+- **Push notifications**, where users can receive notifications of new messages or friend requests using the firebase_messaging package.
```

Comment:
Nothing major, but I would use some formatting to highlight `firebase_messaging`

Reply:
done